perfkitbenchmarker/vm_util.py

```diff
@@ -501,3 +504,51 @@ def GenerateSSHConfig(vms):
     ofp.write(template.render({'vms': vms}))
   logging.info('ssh to VMs in this benchmark by name with: '
                'ssh -F {0} <vm name>'.format(target_file))
+
+
+@contextlib.contextmanager
+def RunDStatIfConfigured(vms):
+  """Installs and runs dstat on 'vms' if FLAGS.dstat is set.
+
+  On exiting the context manager, the results are copied to the run temp dir.
+
+  Args:
+    vms: List of virtual machines.
+
+  Yields:
+    None
+  """
+  if not FLAGS.dstat:
+    yield
+    return
+
+  lock = threading.Lock()
+  pids = {}
+  file_names = {}
+
+  def StartDStat(vm):
+    vm.Install('dstat')
+    dstat_file = os.path.join(VM_TMP_DIR, '{0}-dstat.csv'.format(vm.name))
```

Comment:
I think the filename should be a parameter ... or at least part of the filename.  They way you have this right now the user has no control over the name.  If they start and stop a couple of times it looks like they will only get the latest.

I like passing the name because it will allow us to add processors over DSTAT files more easily.